api/src/resources/user.rs

```diff
@@ -62,6 +62,18 @@ pub enum Identifier {
     Username(Username),
 }
 
+impl FromStr for Identifier {
+    type Err = Error;
+
+    fn from_str(string: &str) -> Result<Self> {
+        if string.chars().all(|c| c.is_digit(16)) {
+            Ok(Identifier::Id(Id(string.into())))
+        } else {
+            Username::from_str(string).map(Identifier::Username)
+        }
+    }
+}
```

Comment:
This is bug prone and surprising imo, e.g. `ada` is a user id; for other resources, e.g. like sources we can figure out if it's an id or full name based on whether there's a slash in the name. 

Can we use a different convention like `id:ada` vs `ada` or something else that uniquely identifies if something is a username or id?

Reply:
You're right. Usernames didn't really work properly anyway, since there is no `get-by-name` route, so I removed username support entirely, can only do id-s now.